### PR TITLE
[GDN] Avoid two per-layer copies in the FlashInfer prefill path

### PR DESCRIPTION
**Pull request by zobinHuang**
[GDN] Avoid two per-layer copies in the FlashInfer prefill path

## Purpose

Each GatedDeltaNet layer pays two extra `[num_tokens, num_v_heads, head_v_dim]` copies per prefill step in the FlashInfer chunk path:

1. The chunk kernel writes a fresh tensor which is then assigned into the pre-allocated `core_attn_out` (`core_attn_out[:n] = out.squeeze(0)`).
2. `_rms_norm_gated_cuda` reshapes the gate `z` to `[n * heads, head_v_dim]`. `z` is a strided slice of the fused `in_proj_qkvz` output, so the reshape materializes a clone (`aten::reshape -> aten::clone` in the profiler).

This PR removes both:

- The prefill rows of `core_attn_out` are passed to `flashinfer.gdn_prefill.chunk_gated_delta_rule` as its `output` buffer, so the kernel writes in place. It is applied only when there is no MTP merge (`spec_sequence_masks is None`); with peeled decode tokens the prefill tail is written directly and only the decode rows are stored afterwards.
- The gated RMSNorm runs on `[n, heads * head_v_dim]` rows with `group_size=head_v_dim` and a per-head weight tiled once and cached, so the strided `z` stays a view. Numerics are unchanged: max abs diff 0.0 against the previous formulation at T=8192 (see below).

Both changes are Python-only and apply to every model using `QwenGatedDeltaNetAttention` with the FlashInfer prefill backend (Qwen3-Next, Qwen3.5, Qwen3.8-Flash-Next).

Related work: #45717 (draft, needs rebase) proposes the same output-buffer idea behind an opt-in env flag and skips mixed decode+prefill batches; it does not address the `z` clone. This PR is unconditional, covers the split batch, and adds the norm change. Happy to coordinate if that PR is revived.

## Test Plan

Norm equivalence (old `reshape(-1, V)` formulation vs. grouped rows):

```bash
python - <<'EOF'
import torch
from vllm.third_party.flash_linear_attention.ops.layernorm_guard import layer_norm_fwd
n, HV, V = 8192, 48, 128
x = torch.randn(n, HV, V, device="cuda", dtype=torch.bfloat16)
z = torch.randn(n, 16384, device="cuda", dtype=torch.bfloat16)[:, 10240:].reshape(n, HV, V)
w = torch.randn(V, device="cuda", dtype=torch.bfloat16) * 0.1 + 1
a = torch.empty_like(x); b = torch.empty_like(x)
layer_norm_fwd(x.reshape(-1, V), w, None, 1e-6, z=z.reshape(-1, V), out=a.reshape(-1, V), group_size=V, norm_before_gate=True, is_rms_norm=True, activation="silu")
layer_norm_fwd(x.view(n, -1), w.repeat(HV), None, 1e-6, z=z.view(n, -1), out=b.view(n, -1), group_size=V, norm_before_gate=True, is_rms_norm=True, activation="silu")
print((a.float() - b.float()).abs().max().item())
EOF
```

Serving benchmark (1x B300 SXM6, TP=1, `Qwen/Qwen3.8-Flash-Next-FP8`):

```bash
vllm serve Qwen/Qwen3.8-Flash-Next-FP8 --served-model-name qwen-flash \
  --tensor-parallel-size 1 --max-model-len 10240 --max-num-seqs 32 \
  --max-num-batched-tokens 8192 --gpu-memory-utilization 0.85 \
  --no-enable-prefix-caching --enable-flashinfer-autotune \
  --limit-mm-per-prompt '{"image":0,"video":0}' \
  --compilation-config '{"mode":0,"cudagraph_mode":"FULL_DECODE_ONLY","cudagraph_capture_sizes":[1,2,4,8,16,32]}'

for case in "8192 1024 1" "8192 1024 8" "2048 128 32" "128 128 1"; do set -- $case
  vllm bench serve --backend vllm --base-url http://127.0.0.1:8000 \
    --model Qwen/Qwen3.8-Flash-Next-FP8 --served-model-name qwen-flash \
    --dataset-name random --random-input-len $1 --random-output-len $2 \
    --random-range-ratio 0 --random-prefix-len 0 --num-prompts $((5*$3)) \
    --num-warmups $((3*$3)) --request-rate inf --max-concurrency $3 \
    --ignore-eos --temperature 0 --seed 42
done
```

Accuracy: GSM8K, full 1319-question test split, 5-shot as multi-turn chat, `enable_thinking=false`, greedy, max 2048 output tokens, 32 concurrent requests against the same server.

## Test Result

Norm equivalence: max abs diff `0.0`; the norm call drops from 185 us to 114 us per layer at T=8192.

Serving (single-run p50; measured on top of two unrelated local decode changes that do not touch the prefill path, so the TTFT delta is attributable to this PR):

| case | before TTFT / TPOT (ms) | after TTFT / TPOT (ms) | tok/s before -> after |
|---|---|---|---|
| 8192 in / 1024 out, c=1 | 238.5 / 5.00 | 236.6 / 5.00 | 191.1 -> 191.3 |
| 8192 in / 1024 out, c=8 | 1153.7 / 9.25 | 1131.7 / 9.08 | 774.9 -> 783.4 |
| 2048 in / 128 out, c=32 | 558.8 / 20.62 | 541.9 / 20.60 | 1263.3 -> 1275.3 |
| 128 in / 128 out, c=1 | 107.1 / 4.96 | 103.6 / 4.96 | 173.6 -> 174.6 |

GSM8K strict exact match: 96.82% (1277/1319) after, 96.82% baseline main.

AI-assisted implementation and testing; reviewed and tested by the submitter.

### CODE REVIEW

File: vllm/model_executor/layers/mamba/gdn/qwen_gdn_linear_attn.py [hunk 1]
```diff
@@ -1852,22 +1875,41 @@ def _rms_norm_gated_cuda(
         x_shape = x.shape
         assert output_gate.shape == x_shape
         assert out.shape == x_shape
-        x_2d = x.reshape(-1, x_shape[-1])
-        output_gate_2d = output_gate.reshape(-1, x_shape[-1])
-        out_2d = out.reshape(-1, x_shape[-1])
+        group_size = (
```
Comment (ZJY0516) [inline vllm/model_executor/layers/mamba/gdn/qwen_gdn_linear_attn.py:1878]: complex but very minor performance improvement😭
